common/txmgr/types/storage_service.go

```diff
@@ -6,30 +6,56 @@ import (
 
 	uuid "github.com/satori/go.uuid"
 
+	"github.com/smartcontractkit/chainlink/v2/common/types"
 	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
 )
 
 // NEWTX, TX, TXATTEMPT will be converted from generic types to structs at a future date to enforce design and type checks
-type TxStorageService[ADDR any, CHAINID any, HASH any, NEWTX any, R any, TX any, TXATTEMPT any, TXID any, TXMETA any] interface {
+//
+//go:generate mockery --quiet --name TxStorageService --output ./mocks/ --case=underscore
+type TxStorageService[
+	// Represents an account address, in native chain format. TODO: Should implement Hashable
+	ADDR any,
+	// Represents a chain id to be used for the chain.
+	CHAINID any,
+	// Represents a unique Tx Hash for a chain
+	TX_HASH types.Hashable[TX_HASH],
+	// Represents a unique Tx Hash for a chain
+	BLOCK_HASH types.Hashable[BLOCK_HASH],
+	NEWTX any,
+	// Represents a onchain receipt object that a chain's RPC returns
+	R any,
+	// Represents a tx object that is used by the TXM
+	// TODO: Remove
```

Comment:
Can't merge with these TODOs unless it has a ticket link

Reply:
Ya, plan to do all these TODOs in a follow-up PR, as a package move for EthTx, EthTxAttempt, without which, the current TODOs can't be done.
I will add ticket: https://smartcontract-it.atlassian.net/browse/BCI-865